students/LouReis/Lesson04/mailroom2.py

```diff
@@ -22,113 +22,123 @@
+    for key,value in sorted(donations.items(),key=lambda i:sum(i[1]),reverse=True):
+        total = 0
+        total = sum(donations[key])
         count = 0
-        amount = 0
-        for y, item_b in enumerate(list):
-            if item_b == donors[x]:
-                name = item_a
-                count = count + 1
-                amount = amount + list[y+1]
-        # The below print statement was for testing unsorted output.
-        # print('{:25} ${:>15} {:>15} ${:>15}'.format(name, amount, count, amount/count))
-        donor_amount.append(amount)
-        donor_count.append(count)
-        donor_average.append(amount/count)
-    print()
-    sort_max(donors, donor_amount, donor_count, donor_average)
+        count = len(donations[key])
+        print ('{:25} ${:>15,.2f} {:>15} ${:>15,.2f}'.format(key, total, count, total/count))
 
 # Below is the main menu function that continues prompting until quit.
 def main_menu(main_prompt,menu_options_dict):
+    import sys
     while True:
         response = input(main_prompt)
         if menu_options_dict[response]() == "Quit":
-            break
+            sys.exit()
 
 # Below are the 4 menu options that are declared in the dict.
-def option_one():
+def donation_report():
     print('\nYou Chose Option 1\n\n')
     print('DONATION SUMMARY REPORT\n\n')
-    total(donations,donor_list)
+    calculate_sort(donations)
 
-def option_two():
+# This Option generates a thank you letter for a new donation and prints to the screen.
+def thanks_letter():
     print('\nYou Chose Option 2\n\n')
     print('Create a Thank You Letter\n\n')
     donor = 'L'
     while donor == 'L':
         donor=input("Enter the full name of the Donor (Type 'L' for a donor list):")
         if donor == 'L':
-            print(donor_list)
-    if donor not in donor_list:
+            print("\n")
+            for key in sorted(donations):
+                print(key)
+            print("\n")
+    if donor not in donations:
         print("You have entered a new donor:", donor)
-        amount = input("Please enter a donation amount '0.00':")
-        amount = float(amount)
-        donor_list.append(donor)
-        donations.append(donor)
-        donations.append(amount)
-    elif donor in donor_list:
+        amount = []
+        donation = input("Please enter a donation amount '0.00':")
+        donation = float(donation)
+        amount.append(donation)
+        donations[donor] = amount
+        # donations.update({donor:amount})
+    else:
         print("You have entered an existing donor:", donor)
-        amount = input("Please enter the donation amount:")
-        amount = float(amount)
-        donations.append(donor)
-        donations.append(amount)
-    print_letter(donor,amount)
+        amount = []
+        donation = input("Please enter the donation amount:")
+        donation = float(donation)
+        amount.append(donation)
+        donations[donor] = donations[donor] + amount
```

Comment:
there is some duplicated code in these blocks, consider this:
```
donation = float(input("Please enter the donation amount:"))
if donor in donations:
    donations[donor].append(donation)
else:
    donations[donor] = [donation]
```
^^ above is more simplified